src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/DataController.cs

```diff
@@ -298,22 +299,34 @@ private DataElement CreateDataElement(HttpRequest request, string elementType, I
                 theStream = request.Body;
                 contentType = request.ContentType;
             }
+            
+            DataElement newData = CreateDataElementHelper(elementType, instance, instanceGuid, creationTime, contentType, contentFileName, fileSize);
 
+            return newData;
+        }
+
+        /// <summary>
+        /// Creates a data element (helper method)
+        /// </summary>
+        /// <returns></returns>
+        internal DataElement CreateDataElementHelper(string elementType, Instance instance, Guid instanceGuid, DateTime creationTime, string contentType, string contentFileName, long fileSize)
+        {
             string dataId = Guid.NewGuid().ToString();
 
             string dataLink = $"{prefix}/instances/{instance.Id}/data/{dataId}";
 
-            // create new data element, store data in blob
+            string user = GetUser(User);
+
             DataElement newData = new DataElement
             {
                 // update data record
                 Id = dataId,
                 ElementType = elementType,
                 ContentType = contentType,
-                CreatedBy = User.Identity.Name,
+                CreatedBy = user,
```

Comment:
Is userName used consistently in the code to identify created by / last changed by?

Reply:
I find that user name is used in the code. I changed variable name to userName so far, to be precise about which field it is spoken of.